### DISCUSSION TITLE
Edge extension side panel populating same data to all tabs

### DISCUSSION THREAD

**Discussion by vigneshayyappan-dev** [microsoft/MicrosoftEdge-Extensions#144] [Discussions]
I have created edge extension with side panel to show tab specific data but same data getting reflected to all the tabs containing side panel. 

 

      

 

Expected : Side panel should show tab specific data. 

 

Example: I have created Sample Extension that displays URL of the page in side panel.  

 

Issue: Seeing same URL in all the tabs side panel. 

Expected Samples:

![image](https://github.com/microsoft/MicrosoftEdge-Extensions/assets/75169125/eece8380-7b2b-4250-a1c1-d77d8f92c5a8)
![image](https://github.com/microsoft/MicrosoftEdge-Extensions/assets/75169125/565a8a40-e9e8-4a37-a1f1-a598c3545aaa)

Current Samples:

![image](https://github.com/microsoft/MicrosoftEdge-Extensions/assets/75169125/831d8437-e2d6-482f-98f0-ec1d99f6dffb)
![image](https://github.com/microsoft/MicrosoftEdge-Extensions/assets/75169125/70234f20-a318-409e-ae56-95bc54677df5)

 

Code samples 

 

Steps, I followed to implement side panel follows below, 

 

Step 1: "permissions": ["sidePanel", "tabs"] 

 

Step 2: Allows users to open the side panel by clicking on the action toolbar icon 

 

``` 

chrome.sidePanel 

  .setPanelBehavior({ openPanelOnActionClick: true }) 

  .catch((error) => console.error(error)); 

 

chrome.tabs.onUpdated.addListener(async (tabId, info, tab) => { 

  if (!tab.url) return; 

  const url = new URL(tab.url); 

  await chrome.sidePanel.setOptions({ 

    tabId, 

    path: 'sidepanel.html', 

    enabled: true 

  }); 

}); 

```

**Comment by ManikanthMSFT** [comment]
Hey @vigneshayyappan-dev, apologies for the delay in responding to your query. Our team has informed us that the issue has been resolved. Could you please confirm if you are still experiencing the problem with the side panel populating the same data across all tabs?

Thank you for your patience.

**Comment by vigneshayyappan-dev** [comment]
Hi Team, Thanks for your response. I will check and update you on this.

Thanks & Regards
Vignesh kumar Ayyappan
